COMPONENT-GUIDELINES.md

```diff
@@ -0,0 +1,293 @@
+[Pipedream](https://pipedream.com) is a low code integration platform that makes it easy to connect APIs remarkably fast. Users can select from thousands of customizable, open source components for hundreds of apps, and orchestrate their execution in workflows. Developers can [contribute](CONTRIBUTING.md) to these open source components on [Github](https://github.com/pipedreamhq/pipedream) by:
+
+- Creating new comonents (sources and actions)
+- Updating exisiting components (e.g., fixing bugs, enhancing functionality)
+- Adding or updating metadata (e.g., descriptions, labels)
+
+Once a PR is merged to the `master` branch of the `pipedreamhq/pipedream` repo, the components are automatically registered and immediately become available to the 150k+ users of the Pipedream platform. 
+
+## Components
+
+Components are [Node.js modules](COMPONENT-API.md#component-structure) that run on Pipedream's serverless infrastructure. They may use Pipedream managed auth for 300+ apps and use most npm packages with no `npm install` or `package.json` required. Pipedream currently supports two types of components — sources and actions.
+
+### Sources
+
+- Emit events that can trigger Pipedream workflows (events may also be consumed outside of Pipedream via API)
+- Emitted event data can be inspected and referenced by steps in the target workflow
+- Can use any of Pipedream's built-in deduping strategies
+- Can be triggered on HTTP requests, timers, cron schedules, or manually
+- May store and retrieve state using the [built-in key-value store](https://github.com/PipedreamHQ/pipedream/blob/master/COMPONENT-API.md#db)
+
+### Actions
+
+- May be used as steps in workflows to perform common functions (e.g., get or modify data in an app)
+- Data returned by actions may be inspected and used in future workflow steps
+
+## Pipedream Registry
+
+The Pipedream registry consists of sources and actions that have been curated for the community. Registered components are verified by Pipedream through the Github PR process, and:
+
+- Can be trusted by end users
+- Follow consistent patterns for usability
+- Are supported by Pipedream if issues arise
+
+Registered components also appear in the Pipedream marketplace and are listed in Pipedream's UI when building workflows.
+
+## Prerequisites
+
+If you're ready to build a component for the Pipedream registry, we recommend starting with our [Quickstart Guide](QUICKSTART.md) and then reviewing the [Component API Reference](COMPONENT-API.md). You will also need the following:
+
+- A free [Pipedream](https://pipedream.com) account
+- A free [Github](https://github.com) account
+- Basic proficiency with Node.js or Javascript
+- Pipedream [CLI](https://pipedream.com/docs/cli/reference/)
+
+Finally, the target app must be integrated with Pipedream. You can explore all apps supported by Pipedream in the [marketplace](https://pipedream.com/explore). If your app is not listed, please [reach out](https://pipedream.com/community/c/dev/11).
```

Comment:
If their app is not listed, can you link them [here](https://github.com/PipedreamHQ/pipedream/issues/new?assignees=&labels=app%2C+enhancement&template=app---service-integration.md&title=%5BAPP%5D), instead?